src/main/scala/nl/biopet/bioconda/BiocondaRecipe.scala

```diff
@@ -0,0 +1,169 @@
+ */
+
+package nl.biopet.bioconda
+
+import org.yaml.snakeyaml.Yaml
+import java.io.{File, PrintWriter}
+
+import scala.collection.JavaConverters._
+import scala.io.Source
+
+class BiocondaRecipe(name: String,
+                     version: String,
+                     command: String,
+                     sourceUrl: String,
+                     sourceSha256: String,
+                     runRequirements: Seq[String],
+                     buildRequirements: Seq[String],
+                     testCommands: Seq[String],
+                     homeUrl: String,
+                     license: String,
+                     summary: String,
+                     defaultJavaOptions: Seq[String],
+                     buildNumber: Int = 0,
+                     notes: Option[String] = None) {
+
+  def fileName: String = sourceUrl.split("/").last
+  def wrapperFilename: String = s"$name.py"
+
+  def stringToFile(string: String, file: File): Unit = {
+    val writer = new PrintWriter(file)
+    writer.write(string)
+    writer.close()
+  }
+
+  def createRecipeFiles(dir: File): Unit = {
+    dir.mkdirs()
+    val buildSh = new File(dir, "build.sh")
+    val meta = new File(dir, "meta.yaml")
+    val wrapper = new File(dir, wrapperFilename)
+    stringToFile(metaYaml, meta)
+    stringToFile(buildScript, buildSh)
+    stringToFile(wrapperScript, wrapper)
+    // Wrapper should be executable
+    wrapper.setExecutable(true)
+  }
+
+  def createRecipe(dir: File): Unit = {
+    dir.mkdirs()
+    createRecipeFiles(dir)
+    //Also create a subdirectory with the version
+    createRecipeFiles(new File(dir, version))
+  }
+
+  def metaYaml: String = {
+    val yaml = new Yaml()
+    val meta: java.util.Map[String, Object] = {
+      mapAsJavaMap(
+        Map(
+          "package" -> Map(
+            "name" -> name,
+            "version" -> version
+          ).asJava,
+          "source" -> Map(
+            "url" -> sourceUrl,
+            "sha256" -> sourceSha256
+          ).asJava,
+          "build" -> Map(
+            "number" -> buildNumber
+          ).asJava,
+          "requirements" -> Map(
+            "run" -> seqAsJavaList[String](runRequirements ++ Seq("python")),
+            "build" -> seqAsJavaList[String](buildRequirements)
+          ).asJava,
+          "about" -> Map(
+            "home" -> homeUrl,
+            "license" -> license,
+            "summary" -> summary
+          ).asJava,
+          "test" -> Map(
+            "commands" -> seqAsJavaList[String](testCommands)
+          ).asJava,
+          "extra" -> Map(
+            "notes" -> notes.getOrElse("")
+          ).asJava
+        )
+      )
+    }
+
+    s"""# Based on OpenJDK recipe in conda-forge
+       |# https://github.com/conda-forge/openjdk-feedstock/blob/master/recipe/meta.yaml
+       |# This file was automatically generated by the sbt-bioconda plugin.
+       |
+     """.stripMargin + "\n" + yaml.dumpAsMap(meta)
```

Comment:
biopet CommonUtils already have method for map -> yaml text

Reply:
great will check that out.

Reply:
common_utils uses scala 2.11. This uses scala 2.12
I get the following errors:
```
- testMetaYaml *** FAILED ***
[info]   java.lang.NoClassDefFoundError: Could not initialize class play.api.libs.json.JsArray$
[info]   at nl.biopet.utils.conversions.package$.anyToJson(package.scala:122)
[info]   at nl.biopet.utils.conversions.package$$anonfun$anyToJson$1.apply(package.scala:121)
[info]   at nl.biopet.utils.conversions.package$$anonfun$anyToJson$1.apply(package.scala:121)
[info]   at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:234)
[info]   at scala.collection.immutable.Map$Map1.foreach(Map.scala:120)
[info]   at scala.collection.TraversableLike.map(TraversableLike.scala:234)
[info]   at scala.collection.TraversableLike.map$(TraversableLike.scala:227)
[info]   at scala.collection.AbstractTraversable.map(Traversable.scala:104)
[info]   at nl.biopet.utils.conversions.package$.anyToJson(package.scala:121)
[info]   at nl.biopet.utils.conversions.package$$anonfun$mapToJson$1.apply(package.scala:111)
[info]   ...
```